File: debounce-plugin/src/main/java/com/sunyuan/click/debounce/entity/MethodEntity.java
package com.sunyuan.click.debounce.entity;

import com.android.annotations.NonNull;

import java.io.Serializable;

/**
 * author : Sy007
 * date   : 2020/11/28
 * desc   :
 * version: 1.0
 */
public class MethodEntity implements Serializable {
    // 方法名
    private String methodName;
    // 方法描述
    private String methodDesc;
    // 方法所在的接口
    private String interfaceName;

    private String name;


    public MethodEntity(@NonNull String name) {
        this.name = name;
    }


    public MethodEntity(String methodName, String methodDesc, String interfaceName) {
        this.methodName = methodName;
        this.methodDesc = methodDesc;
        this.interfaceName = interfaceName;
    }



    public String getMethodName() {
        return methodName;
    }

    public String getMethodDesc() {
        return methodDesc;
    }

    public String getInterfaceName() {
        return interfaceName;
    }

    /*support Gradle DLS*/
    public void methodName(String methodName) {
        this.methodName = methodName;
    }

    public void methodDesc(String methodDesc) {
        this.methodDesc = methodDesc;
    }

    public void interfaceName(String interfaceName) {
        this.interfaceName = interfaceName;
    }
}

File: debounce-plugin/src/main/java/com/sunyuan/click/debounce/entity/MethodEntity.kt
package com.sunyuan.click.debounce.entity

import java.io.Serializable

/**
 * author : Sy007
 * date   : 2020/11/29
 * desc   :
 * version: 1.0
 */

open class MethodEntity : Serializable {
    var methodName: String = ""

    var methodDesc: String = ""

    var owner: String = ""

    /**
     * lambda method access flags.
     */
    var access: Int = -1

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is MethodEntity) return false
        if (owner != other.owner) return false
        if (methodName != other.methodName) return false
        if (methodDesc != other.methodDesc) return false
        return true
    }

    override fun hashCode(): Int {
        var result = methodName.hashCode()
        result = 31 * result + methodDesc.hashCode()
        result = 31 * result + owner.hashCode()
        return result
    }

    fun nameWithDesc() = methodName + methodDesc

}
